### ISSUE TITLE
Large number of metadata objects: what can be a reason?

### ISSUE THREAD

**Issue by nchaly** [aspnet/EntityFramework6#199]
Performing memory profiling, found quite a number of objects related to EF metadata, mostly `TypeUsage` and `MetadataProperty` .

What could be a reason and how to diagnose this issue? 
I assume that it depends on number of tables/columns/relations, but looks like there is no linear dependency. 

Here is the test for just two tables - about 800 TypeUsage objects.
![mc-test-app-2-tables](https://cloud.githubusercontent.com/assets/2665273/22792112/b4e250a8-eefc-11e6-909d-e1ab0646246c.PNG)

Here is the test for 50 tables - about 10K TypeUsages:
![mc-ca-app-25-tables](https://cloud.githubusercontent.com/assets/2665273/22792170/eec0988e-eefc-11e6-93b5-2541060ef24d.PNG)

Production mode with several apps in worker process:
![ef-30-group-by-assembly2](https://cloud.githubusercontent.com/assets/2665273/22791620/15384324-eefb-11e6-8c38-ed65b0062982.png)

EF version: 6.1.3.

**Comment by divega** [comment]
@nchaly in case it helps: EF6 canonical functions and providers specific functions are described using the same metadata objects you are counting here and those get loaded into memory even before a specific model is loaded. In other words, there will be a minimum of metadata objects loaded into memory even for an empty model.

Also, AFAIR, there are `TypeUsage` objects for each column and each property and each parameter in the model.

**Comment by nchaly** [comment]
@rowanmiller I don't think the question has the proper answer.

**Comment by rowanmiller** [comment]
@nchaly can you share a project or code listing that we can use to reproduce the issue?

**Comment by nchaly** [comment]
Hi again. The code in question was a part of IP and I had to create something from scratch. It took a while, but now it's ready - https://github.com/nchaly/ef6-metadata-test

My goal was to make something simple and "all-inclusive": I've taken a Northwind model as a sample; using ClrMD I was able to count objects without need to get an external profiler. 

Here is the chart built on sample data - https://raw.githubusercontent.com/nchaly/ef6-metadata-test/master/Chart.PNG. Growth rate for TypeUsage is quite high, as well as for entire System.Data.Entity.Metadata.Edm namespace.
